src/cljs/commiteth/handlers.cljs

```diff
@@ -39,6 +40,20 @@
    (println "redirecting to" path)
    (set! (.-pathname js/location) path)))
 
+(reg-fx
+ :persist-bounty-filters-in-query
+ (fn [{:keys [bounty-filters]}]
+   (let [query
+         (reduce-kv
+          #(if (nil? %3)
+             %1
+             (let [param-name (ui-model/bounty-filter-type->query-param %2)
+                   param-value (ui-model/bounty-filter-value->query-param %2 %3)]
+               (assoc %1 param-name param-value)))
+          {}
+          bounty-filters)]
+     (routes/nav! :bounties {} query))))
```

Comment:
```clojure
;; untested
(->> bounty-filters
     (remove (comp nil? val))
     (map (fn [[k v]]
            [(ui-model/bounty-filter-type->query-param k)
             (ui-model/bounty-filter-value->query-param k v)]))
     (into {}))
```
Maybe something like this is a bit easier to read? Also no need to bind via `let` if the value is only used once.

Reply:
Actually regarding use of `let` — sometimes it's good to explicitly declare what something is. So if you feel like binding stuff to a name here adds value then please keep it :)

Reply:
Keeping the `let` but using your implementation. I was pretty unhappy with what I came up with. This is so cool, thanks for the feedback!